plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -208,26 +213,48 @@ class CodeTransformChatController(
-        }
+        codeTransformChatHelper.addNewMessage(buildUserSelectionSummaryChatContent(moduleName))
 
-        // this should never throw the RuntimeException since invalid JDK case is already handled in previous validation step
-        val moduleJdkVersion = ModuleUtil.findModuleForFile(moduleVirtualFile, context.project)?.tryGetJdk(context.project)
-        logger.info { "Found project JDK version: ${context.project.tryGetJdk()}, module JDK version: $moduleJdkVersion. Module JDK version prioritized." }
-        val sourceJdk = moduleJdkVersion ?: context.project.tryGetJdk() ?: throw RuntimeException("Unable to determine source JDK version")
+        val sourceJdk = getSourceJdk(moduleVirtualFile)
 
         val selection = CustomerSelection(
             moduleVirtualFile,
             sourceJdk,
-            JavaSdkVersion.JDK_17
+            JavaSdkVersion.JDK_17,
         )
 
+        // Create and set a session
+        val session = codeModernizerManager.createCodeModernizerSession(selection, context.project)
+        codeModernizerManager.codeTransformationSession = session
+
         // Publish metric to capture user selection before local build starts
         telemetry.submitSelection("Confirm", selection)
 
-        codeModernizerManager.runLocalMavenBuild(context.project, selection)
+        codeTransformChatHelper.run {
+            addNewMessage(buildUserInputSkipTestsFlagChatIntroContent())
+            addNewMessage(buildUserInputSkipTestsFlagChatContent())
+        }
+    }
+
+    override suspend fun processCodeTransformConfirmSkipTests(message: IncomingCodeTransformMessage.CodeTransformConfirmSkipTests) {
+        val customBuildCommand = when (message.skipTestsSelection) {
+            message("codemodernizer.chat.message.skip_tests_form.skip") -> MAVEN_BUILD_SKIP_UNIT_TESTS
+            else -> MAVEN_BUILD_RUN_UNIT_TESTS
+        }
+        codeTransformChatHelper.addNewMessage(buildUserSkipTestsFlagSelectionChatContent(message.skipTestsSelection))
+        codeTransformChatHelper.addNewMessage(buildCompileLocalInProgressChatContent())
+        codeModernizerManager.codeTransformationSession?.let {
+            it.sessionContext.customBuildCommand = customBuildCommand
+            codeModernizerManager.runLocalMavenBuild(context.project, it)
+        }
+    }
+
+    private fun getSourceJdk(moduleConfigurationFile: VirtualFile): JavaSdkVersion {
+        // this should never throw the RuntimeException since invalid JDK case is already handled in previous validation step
+        val moduleJdkVersion = ModuleUtil.findModuleForFile(moduleConfigurationFile, context.project)?.tryGetJdk(context.project)
+        logger.info { "Found project JDK version: ${context.project.tryGetJdk()}, module JDK version: $moduleJdkVersion. Module JDK version prioritized." }
+        val sourceJdk = moduleJdkVersion ?: context.project.tryGetJdk() ?: throw RuntimeException("Unable to determine source JDK version")
```

Comment:
surprised the linter didn't kick in

```suggestion
        val sourceJdk = moduleJdkVersion ?: context.project.tryGetJdk() ?: error("Unable to determine source JDK version")
```